tests/test_sims.py

```diff
@@ -575,6 +575,25 @@ def test_sbm_multiwt_undirected_loopy(self):
         self.assertTrue(A.shape == (np.sum(self.n), np.sum(self.n)))
         pass
 
+    def test_sbm_dc_undirected_loopy(self):
```

Comment:
should have more tests than this I think. Most important would probably be testing with a weighting function (unless I'm missing something?) rather than a scalar value. A directed case might also be good, since it's more general.

Reply:
I believe I meant to make it directed in the first place so that change has been made. When I talked to j1c about this, his thought was only this was test necessary. The degree-correct code only modifies the indices to be assigned. Everything else before/after in the pipeline is untouched, such as the weights, and the expected number of edges in each block is the same.

Reply:
I don't agree that we don't need a function-passing test, that part of your code is definitely different. For example, the following doesn't work for me:

`A = sbm(self.n, self.Psy, directed=True, loops=True, dc=np.random.poisson, dcargs={"lam":1.0})`
  
is this the expected usage? Also, are dcargs supposed to be mandatory (eg could I just use a poisson lambda=1 with no args)?

Reply:
Ah that weighting function. Yes, tests for that seem appropriate. I ran that code and the error only comes up sometimes so it may be tricky to test for, will need to think about. But it's a simple change to make dcargs optional so will do that too.